HardFork.md

```diff
@@ -0,0 +1,69 @@
+
+You will need to build namada from source using [#2937](https://github.com/anoma/namada/pull/2937), which is the branch `brent/test-migration-from-0.31.9`. This is also the same branch that `namadan-0.31.9-MIG` is built from.
+
+Given that you have built namada from source on this branch, follow these steps to produce `migrations.json`:
+
+1. Go to the root directory of your Namada built from source
+2. Run `namadan-0.31.9-MIG ledger query-db --db-key conversion_state --hash 05E2FD0BEBD54A05AAE349BBDE61F90893F09A72850EFD4F69060821EC5DE65F --db-column-family state > conversion_state.txt`
+3. Copy the new `vp.wasm` and `checksums.json` files from `<insert_link>` into the `wasm/` directory relative to your current path (`namada/wasm/`).
+4. Run `cargo run --example make-db-migration`, which should produce `migrations.json` in the current directory
+
+Additionally, the Namada team will provide a `migrations.json` file on Github at `<insert_URL>` along with a `shasum` of the file. The team will only be able to generate this file once the chain has halted at the planned hard fork block height `X`, as the contents of this file are dependent on the state at that block height.
+
+While the hosted `migrations.json` file may be downloaded and used, the team encourages operators to try to produce the file themselves and verify that the `shasum` of your own file matches that of the hosted one. You can do this by running `shasum migrations.json`, which will give you a hash.
+
+### State migration
+
+Now that you have `migrations.json`:
+
+1. Run `namadan-0.32.0-MIG ledger update-db --path migrations.json --block-height X`. This will update the DB and try to progress the ledger by two blocks before suspending itself again. The ledger will only be able to progress the two blocks and finish this step if enough voting power is online with this command in execution. After two blocks, the ledger will halt. You may need to kill it yourself with `CTRL+C`, or it may exit for you.
+2. Once the ledger is stopped after updating the DB, run the ledger as normal with the new binaries: `namadan-0.32.0 ledger run`.
+
+## Shielded syncing after the hard fork
+
+The hard fork will also make properly scanning for old MASP notes by running `namadac shielded-sync` a bit trickier. After the hard fork, the client will no longer be able to make RPC calls to cometBFT to query for MASP notes before block height `X`, when the hard fork takes place. There are several solutions to help ameliorate this issue:
+
+1. Some nodes that are still running v0.31.9 of the software are listed at `<insert_URL>`. You can query these nodes with `namadac shielded-sync` to scan for all MASP notes up until height `X`.
+2. In v0.32.0, `shielded-sync` has a new option `--from-height` that will start the MASP note scanning at the given height. When you query from a v0.32.0 node, please for at least the first time, use `namadac shielded-sync --from-height X`.
```

Comment:
The precise conditions to be able to remove the `--from-height` flag are as follows:
 1. The user must have keys in their shielded context
 2. They successfully scan a masp tx with this command (if there are none, they will keep having to pass `--from-height X`.)